### PR TITLE
use backbone + marionette to render the list of rooms

### PR DESCRIPTION
**Pull request by ChristophWurst**
This simplifies the process of keeping the list in sync and rendering the list in the left sidebar.

In the future we can then easily order the list by a certain criteria (e.g. last used) and add new channels right away (Backbone will take care of persisting it via the REST api).

### CODE REVIEW

**Comment by ChristophWurst** [comment]
- [ ] TODO: use backbone sync when creatinga new room too

**Comment by jancborchardt** [comment]
@ChristophWurst so should we review or wait for the todo? Sounds like review ;)

File: js/app.js [hunk 1]
```diff
@@ -122,11 +132,25 @@
 				OCA.SpreedMe.Rooms.ping();
 			}, 5000);
 		},
+		syncRooms: function(options) {
```
Comment (LukasReschke) [inline js/app.js]: 135:23  error  'options' is defined but never used  no-unused-vars

**Comment by jancborchardt** [comment]
Tested and doesn’t really break anything :+1:

**Comment by ChristophWurst** [comment]
just pushed a fix for the TODO ;-) Please re-review

**Comment by ChristophWurst** [comment]
fixed lukas' nitpick too

**Comment by LukasReschke** [comment]
Merge conflict 🙈

**Comment by ChristophWurst** [comment]
resolved 🚀

**Pull request merged**